sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -30,21 +30,48 @@ import org.apache.spark.sql.types._
 import org.apache.spark.unsafe.types.{CalendarInterval, UTF8String}
 
 /**
+ * Common base class for time zone aware expressions.
+ */
+trait TimeZoneAwareExpression extends Expression {
+
+  /** the timezone ID. */
+  def timeZoneId: String
+
+  /**
+   * Returns true if the timezone ID is resolved.
+   * If not resolved, the analyzer [[ResolveTimeZone]] will resolve with session local timezone.
+   */
+  def timeZoneResolved: Boolean = timeZoneId != null
```

Comment:
do you mean we may allow users to set timezone id for every `TimeZoneAwareExpression` in the future?